File: solution/implementation/1212/main.py
# Authored by : tony9402
# Co-authored by : -
# Link : http://boj.kr/7d0d2bcb568d4c3d86770e520838ef65
import sys

def input():
    return sys.stdin.readline().rstrip()

def change(num, first = False):
    ret = ''
    while num:
        ret += chr(num % 2 + 48)
        num //= 2
    while len(ret) < 3:
        ret += '0'

    idx = 3
    if first:
        while idx > 1 and ret[idx - 1] == '0':
            idx -= 1
    return ret[:idx][::-1]

N = input()
isFirst = True
for i in range(len(N)):
    print(change(int(N[i]), isFirst),end='')
    isFirst = False

File: solution/implementation/1212/main.cpp
// Authored by : tony9402
// Co-authored by : -
// Link : http://boj.kr/9aebf3cefd45466aa4c24dae72837024
#include<bits/stdc++.h>

using namespace std;

string convert(char s, bool fillzero=false){
    int n = s - 48;
    string ret = "";
    while(n){
        ret += (n % 2) + '0';
        n /= 2;
    }
    while(fillzero && (int)ret.size() < 3) ret += '0';
    reverse(ret.begin(), ret.end());
    return ret;
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(0);

    string s;cin >> s;
    if(s == "0") {
        cout << 0;
        return 0;
    }
    string ans="";
    for(int i=0;i<(int)s.size();i++){
        ans += convert(s[i], i != 0);
    }
    cout << ans;

    return 0;
}

